guava/src/com/google/common/escape/ArrayBasedCharEscaper.java

```diff
@@ -46,7 +48,7 @@
   // The replacement array (see ArrayBasedEscaperMap).
   private final char[][] replacements;
   // The number of elements in the replacement array.
-  private final int replacementsLength;
+  private final @LTEqLengthOf("replacements") int replacementsLength;
```

Comment:
Here and throughout, I'd use `@LengthOf` instead of `LTEqLengthOf`, since that more accurately represents what `replacementsLength` actually is.